openmdao/test_suite/test_examples/beam_optimization/components/interp.py

```diff
@@ -0,0 +1,120 @@
+"""
+Simple B-spline component for interpolation of beam thickness as a function of length along the beam.
```

Comment:
Well, I considered this a one-off and got some interpolant code from John. If we want to add this as a feature to OpenMDAO, shouldn't we figure out what we want to provide users in terms of API and the underlying methods, and then code something to that, rather than taking what comes along?